### ISSUE TITLE
[Bug] make create-demo-cluster问题

### ISSUE THREAD

**Issue by paymanfu** [apache/cloudberry#746]
### Apache Cloudberry version

main branch

### What happened

![image](https://github.com/user-attachments/assets/a1960764-03ee-4457-9ac4-6beebdd078c0)
执行 make create-demo-cluster以后，由于之前没有ssh localhost，这里就一直打点号

### How to reproduce

不应该有..... 这些输出，没有给用户输入yes or no的间隙

### Operating System

 	Welcome to Huawei Cloud Service  [root@2x86-64-6d1e ~]# cat /etc/os-release NAME="CentOS Linux" VERSION="7 (Core)" ID="centos" ID_LIKE="rhel fedora" VERSION_ID="7" PRETTY_NAME="CentOS Linux 7 (Core)" ANSI_COLOR="0;31" CPE_NAME="cpe:/o:centos:centos:7" HOME_URL="https://www.centos.org/" BUG_REPORT_URL="https://bugs.centos.org/"  CENTOS_MANTISBT_PROJECT="CentOS-7" CENTOS_MANTISBT_PROJECT_VERSION="7" REDHAT_SUPPORT_PRODUCT="centos" REDHAT_SUPPORT_PRODUCT_VERSION="7"

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/cloudberry/blob/main/CODE_OF_CONDUCT.md).

**Comment by yihong0618** [comment]
for this gpdb have a better solution by higuoxing if its the same host we can skip this, 
more: https://github.com/greenplum-db/gpdb-archive/commit/237eac473ba0f5070ecdbfedb4937c11850139bd

also it can speed up test in local @avamingli

**Comment by avamingli** [comment]
Hi, @paymanfu Could you please rewrite in English so other engineers could participant, thanks!

**Comment by avamingli** [comment]
> for this gpdb have a better solution by higuoxing if its the same host we can skip this, more: [greenplum-db/gpdb-archive@237eac4](https://github.com/greenplum-db/gpdb-archive/commit/237eac473ba0f5070ecdbfedb4937c11850139bd)
> 
> also it can speed up test in local @avamingli

@yihong0618 Thanks! Your advise do save a lot of time.

**Comment by avamingli** [comment]
> > for this gpdb have a better solution by higuoxing if its the same host we can skip this, more: [greenplum-db/gpdb-archive@237eac4](https://github.com/greenplum-db/gpdb-archive/commit/237eac473ba0f5070ecdbfedb4937c11850139bd)
> > also it can speed up test in local @avamingli
> 
> @yihong0618 Thanks! Your advise do save a lot of time.

I'm not familiar with Python codes, but I believe others might be able to help.

**Comment by Smyatkin-Maxim** [comment]
It's been this way forever, you need to setup some ssh keys for GP segments to communicate. Something like this works for linux:
```
ssh-keygen
cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
chmod 600 ~/.ssh/authorized_keys
```
Depending on your setup you might also `ssh-add` your key.

But, indeed, we better just cherrypick the GP7 commit, it's a good one.

**Comment by edespino** [comment]
I actually believe the user does have an ssh key created and in the proper location. The error would be different if it didn't exist. I believe the issue is that the initial host verification has not taken place. One can use ssh-keyscan to help with this. Essentially run the following replacing HOSTNAME for all hosts in the cluster. This should fix your issue:

`ssh-keyscan HOSTNAME >> ~/.ssh/known_hosts`

TL;DR ssh-keyscan is a utility that helps gather the public SSH host keys of a number of hosts. How does it helps with the initial SSH connection to a system? By building .ssh/known_hosts: ssh-keyscan collects the public keys of remote hosts and adds them to the .ssh/known_hosts file. This file is used by SSH to verify the identity of the remote host, preventing man-in-the-middle attacks.

**Comment by avamingli** [comment]
> I actually believe the user does have an ssh key created and in the proper location. The error would be different if it didn't exist. I believe the issue is that the initial host verification has not taken place. One can use ssh-keyscan to help with this. Essentially run the following replacing HOSTNAME for all hosts in the cluster. This should fix your issue:
> 
> `ssh-keyscan HOSTNAME >> ~/.ssh/known_hosts`
> 
> TL;DR ssh-keyscan is a utility that helps gather the public SSH host keys of a number of hosts. How does it helps with the initial SSH connection to a system? By building .ssh/known_hosts: ssh-keyscan collects the public keys of remote hosts and adds them to the .ssh/known_hosts file. This file is used by SSH to verify the identity of the remote host, preventing man-in-the-middle attacks.

+1, Hi, @paymanfu  have you done these steps as the official doc? 
we have all steps including ssh in https://github.com/apache/cloudberry/blob/main/deploy/build/README.Linux.md

>useradd gpadmin  # Creates gpadmin user
su - gpadmin  # Uses the gpadmin user
ssh-keygen  # Creates SSH key
cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
chmod 600 ~/.ssh/authorized_keys
exit

>  so that you can run ssh localhost without a password.

In GPDB, the xxxxOS.README files are  listed under the src root dir, but CBDB moved those into deep dir `deploy/build/`.
I don't know the reason, perhaps we should move them back to make it easy for new comers find them.

**Comment by avamingli** [comment]
> But, indeed, we better just cherrypick the GP7 commit, it's a good one.

+1

**Comment by yihong0618** [comment]
Just cherry-pick two commit refer to this.
